### PR TITLE
Code culture wip

### PR DESCRIPTION
**Pull request by timcharper**

### CODE REVIEW

File: docs/docs/code-culture.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+This default speaks to the desire to write code that can be understood without needing an IDE. Marathon is written in Scala, so our defaults are specific as such.
+
+### On Implicits
+
+Implicits are an important feature in Scala and are critical for the implementation of type-classes and DSLs.
+
+Prefer:
+
+* Explicit conversion over implicit
+* Explicit passing over implicit
```
Comment (jeschkies) [inline docs/docs/code-culture.md]: Explicit imports of Implicits 🙂

File: docs/docs/code-culture.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+### On Playing Code Golf
+
+Code golf is a game in which you implement some program in as few characters as possible.
+
+We prefer not to play this game when working on Marathon. Instead, we focus on removing noise (boilerplate), while preserving valuable signal pertaining to the problem at hand.
+
+### On Imports
+
+We prefer, almost always, that implicits go at the top of the file.
```
Comment (jeschkies) [inline docs/docs/code-culture.md]: Imports or implicits?

File: docs/docs/code-culture.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+Prefer:
+
+```
+import org.company.{Stringifier, Demuxer}
+```
+
+Over:
+
+```
+import org.company._
```
Comment (jeschkies) [inline docs/docs/code-culture.md]: We probably should spell out to avoid wild card imports.

File: docs/docs/code-culture.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+Fail loudly. Fail proudly.
+
+## Let it Crash
+
+We prefer to focus on state recovery, and not graceful tear down. We'd prefer to just crash and restart, rather than implement many complex special error handlers.
+
+## Don't Panic
+
+We prefer to that Exceptions are used only when something goes unexpectedly wrong.
```
Comment (jeschkies) [inline docs/docs/code-culture.md]: If I remember correctly we said that exceptions should be considered panics unless they are throw by the library.

File: docs/docs/code-culture.md (hunk 2 above)
Comment (timcharper) [reply docs/docs/code-culture.md]: ✅  imports

File: docs/docs/code-culture.md (hunk 3 above)
Comment (timcharper) [reply docs/docs/code-culture.md]: ✅

File: docs/docs/code-culture.md (hunk 4 above)
Comment (timcharper) [reply docs/docs/code-culture.md]: Yeah, this is hard to describe. I tried to paint the picture with examples.

File: docs/docs/code-culture.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+Prefer:
+
+* Explicit conversion over implicit
+* Explicit passing over implicit
+
+
+For our case, if implicits don't help improve the design of code, we don't use them. We don't use implicit conversions. We don't do multi-priority implicits. When type classes help improve the design of code, we use implicits for type classes (IE serialization logic).
+
+We prefer implicits to be used contextual values, such as execution contexts. However, we prefer to be conservative, and pass the value explicitly for things like validators, authenticators, and normalizers.
```
Comment (alenkacz) [inline docs/docs/code-culture.md]: 👍

**Pull request merged**